### PR TITLE
Remove flipt in favour of lib implementation

### PR DESCRIPTION
**Pull request by danvixent**

### CODE REVIEW

File: api/types/types.go [hunk 1]
```diff
@@ -12,10 +13,11 @@ import (
 type ContextKey string
 
 type APIOptions struct {
-	DB     database.Database
-	Queue  queue.Queuer
-	Logger log.StdLogger
-	Tracer tracer.Tracer
-	Cache  cache.Cache
-	Authz  *authz.Authz
+	FlagCtrl *fflag.Controller
```
Comment (jirevwe) [inline api/types/types.go:16]: Why not name it `Fflag`?

File: config/config.go [hunk 2]
```diff
@@ -226,8 +226,10 @@ type TypesenseConfiguration struct {
 }
 
 type FeatureFlagConfiguration struct {
-	Type  FeatureFlagProvider `json:"type" envconfig:"CONVOY_FEATURE_FLAG_TYPE"`
-	Flipt FliptConfiguration  `json:"flipt"`
+	Experimental bool `json:"experimental" envconfig:"CONVOY_EXPERIMENTAL_FEATURES"`
+	Alpha        bool `json:"alpha" envconfig:"CONVOY_ALPHA_FEATURES"`
+	Beta         bool `json:"beta" envconfig:"CONVOY_BETA_FEATURES"`
+	GA           bool `json:"ga" envconfig:"CONVOY_GA_FEATURES"`
```
Comment (jirevwe) [inline config/config.go:232]: Isn't GA implied when nothing is passed?

File: cmd/main.go [hunk 3]
```diff
@@ -46,6 +47,11 @@ func main() {
 	var dbPassword string
 	var dbDatabase string
 
+	var experimental bool
+	var alpha bool
```
Comment (subomi) [inline cmd/main.go:51]: @danvixent You forgot to take these values out.

File: datastore/models.go [hunk 4]
```diff
@@ -1457,3 +1457,19 @@ func (p *Password) Matches() (bool, error) {
...
+
+const (
+	ExperimentalFlagType FlagType = "experimental"
+)
+
+func (ft FlagType) String() string {
+	return string(ft)
+}
+
+type Flag struct {
```
Comment (subomi) [inline datastore/models.go:1471]: @danvixent You forgot to remove these as well.

File: datastore/models.go [hunk 5]
```diff
@@ -1457,3 +1457,19 @@ func (p *Password) Matches() (bool, error) {
 
 	return true, err
 }
+
+type FlagType string
```
Comment (subomi) [inline datastore/models.go:1461]: @danvixent Please move this to the `fflag` package.

File: sql/1696938095.sql [hunk 6]
```diff
@@ -0,0 +1,17 @@
+-- +migrate Up
+CREATE TABLE IF NOT EXISTS convoy.feature_flags (
```
Comment (subomi) [inline sql/1696938095.sql:2]: @danvixent You forgot to remove this as well.

File: internal/pkg/fflag/fflag.go [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+var features = map[string]datastore.FlagType{}
+
+type FFlag struct{}
+
+func NewFFlag() *FFlag {
+	return &FFlag{}
+}
+
+func (c *FFlag) CanAccessFeature(key string, cfg *config.Configuration) bool {
```
Comment (subomi) [inline internal/pkg/fflag/fflag.go:16]: @danvixent The logic here is not correct. We should compare if the retrieved `flagType` is less than or equal to the user's selection.

**Review by subomi — CHANGES_REQUESTED** [review]

**Review by jirevwe — APPROVED** [review]

**Review by subomi — APPROVED** [review]

**Pull request merged**